pkg/exporter/process.go

```diff
@@ -83,13 +95,21 @@ func InitExportingProcess(input ExporterInput) (*ExportingProcess, error) {
 	var err error
 
 	// Resolve TCP and UDP addresses
+	collectorAddress, err := util.ResolveDNSAddress(input.CollectorAddress)
+	if err != nil {
+		return nil, err
+	}
 	if input.CollectorProtocol == "tcp" {
-		collectorAddr, err = net.ResolveTCPAddr(input.CollectorProtocol, input.CollectorAddress)
-		if err != nil {
-			return nil, err
+		if input.IsEncrypted {
+			collectorAddr = &tlsAddress{input.CollectorAddress}
```

Comment:
do you need this if block? Just use input.CollectorAddress directly in TLS dial in L124? 
If encryption is required, add a check to make sure input.CollectorAddress is not IP and if not, throw an error.

Reply:
Actually we don't need this block and tlsAddress above. Removed and changed L124. Thanks!